pkg/cloudprovider/providers/azure/azure_controller_vmss.go

```diff
@@ -71,8 +71,12 @@ func (ss *scaleSet) AttachDisk(isManagedDisk bool, diskName, diskURI string, nod
 
 	ctx, cancel := getContextWithCancel()
 	defer cancel()
-	klog.V(2).Infof("azureDisk - update(%s): vm(%s) - attach disk(%s)", nodeResourceGroup, nodeName, diskName)
 
+	// Invalidate the cache right after updating
+	key := buildVmssCacheKey(nodeResourceGroup, ss.makeVmssVMName(ssName, instanceID))
+	defer ss.vmssVMCache.Delete(key)
```

Comment:
here is the PR for this dirty cache issue, originally whether succeed or not, this PR clean the cache anyway.
I can move this clean cache operation `defer ss.vmssVMCache.Delete(key)` to conditions when only error happens, would that solve your issue @khenidak ?I can build one hot fix image if you wan to try this fix.
cc@feiskyer